sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CollectionQuerySimplification.cs

```diff
@@ -467,7 +433,7 @@ private static bool MethodIsNotUsingPredicate(IMethodSymbol methodSymbol, Invoca
         {
             var arguments = GetReducedArguments(methodSymbol, invocation);
 
-            return !arguments.Any() && MethodNamesWithPredicate.Contains(methodSymbol.Name);
+            return arguments.Count == 0 && MethodNamesWithPredicate.Contains(methodSymbol.Name);

```

Comment:
You are really obsessed with performance, eh? :) I am curious if it will be much faster in real world if we first check arguments.Count and then call GetReducedArguments...